packages/core/src/codewhisperer/activation.ts

```diff
@@ -337,7 +342,26 @@ export async function activate(context: ExtContext): Promise<void> {
             [...CodeWhispererConstants.securityScanLanguageIds],
             SecurityIssueCodeActionProvider.instance
         ),
-        vscode.commands.registerCommand('aws.amazonq.openEditorAtRange', openEditorAtRange)
+        vscode.commands.registerCommand('aws.amazonq.openEditorAtRange', openEditorAtRange),
+        auth.regionProfileManager.onDidChangeRegionProfile(() => {
+            // validate user's still has access to the selected customization
+            const selectedCustomization = getSelectedCustomization()
+            // no need to validate base customization which has empty arn
+            if (selectedCustomization.arn.length > 0) {
+                getAvailableCustomizationsList()
+                    .then((customizations) => {
+                        const r = customizations.find((it) => it.arn === selectedCustomization.arn)
+                        if (!r) {
+                            void switchToBaseCustomizationAndNotify().then()
```

Comment:
`void` silences any errors, so the `.catch()` would not be called. and the `.then()` isn't needed in any case?

if you want to silence errors, the `.catch()` below is enough.

Reply:
> he void is the least preferred way to do that, because it will let the unresolved promise rejection go into the void and be handled at a random point by nodejs.

isee, didn't know this before thanks for pointing out Justin

Reply:
so is thiss the correct way to do async call in the `.then()` callback 
https://github.com/aws/aws-toolkit-vscode/pull/7049/commits/35a5045629de765037c3751c2c386f885764fb54

Reply:
see https://github.com/aws/aws-toolkit-vscode/pull/7049#discussion_r2045701021